src/marlin_stubs/pause/pause.cpp

```diff
@@ -625,6 +627,95 @@ bool Pause::wait_or_stop() {
     return false;
 }
 
+// TODO: Implement non-blocking homing, put it in homing.cpp and replace this
```

Comment:
I would propose to create wrapper function, which would parse the Gcode and then call function G28_no_parser(arguments) and this function could be called from other parts of the FW, even in other Gcodes. The same way is done M1400 in M70X.cpp file.

Reply:
I agree.